### ISSUE TITLE
Feature Request: Add function to set key value for level logger

### ISSUE THREAD

**Issue by jpudney** [go-kit/kit#503]
Yo! Recently I needed to change the value of the level key to ensure that our current log storage could slurp up the logs of a new service that's been developed with go-kit/kit/log as it's logger.

I've committed the [changes](https://github.com/jpudney/kit/commit/bff888cf052b86a5a7849a90a6738c3825634c7c) to a fork and wondered if you wanted them?

**Comment by ChrisHines** [comment]
Short answer: No, I don't think we want to provide a global `level.SetKey` function.

The commit you linked says:

>The current key value "level" is hard coded to ensure that no unecessary
allocations are made following the instantiation of the logger. However,
there are times when this value needs to be somethin else. This changes
provides a method for doing so.

That is not the whole truth. Avoiding an allocation is a nice benefit of having a global key value, but it was not what drove the design.

The primary reason there is no `level.SetKey` function is because it opens the door for a data race if it is called concurrently with `level.Error` or the like. We could protect the key value with a mutex, but that could then become an unnecessary bottleneck imposed on all leveled log calls for all package clients.

The inability to change the key used by `level.Error` and its siblings was a tradeoff we accepted in order to allow leveled logging to build cleanly on top of the core log package without introducing a new `level.Levels` type to hold configuration information as the deprecated_levels package does. We found over time that the approach taken in deprecated_levels forced client code to make a choice between accepting a `log.Logger` or a `levels.Levels`. This dichotomy resulted in impedance mismatches between code on either side of this divide that we did not like.

Given the above, I believe the best way to support a custom level key without maintaining a fork of the level package, is to add use a `log.Logger` decorator that changes the level key prior to the `keyvals` being formatted for output. For example:

```go

func main() {
	l := log.NewLogfmtLogger(os.Stdout)
	l = setLevelKey(l, "myLevelKey")
	l = level.NewFilter(l, level.AllowInfo())

	level.Info(l).Log("foo", "bar")
}


func setLevelKey(logger log.Logger, key interface{}) log.Logger {
	return log.LoggerFunc(func(keyvals ...interface{}) error {
		for i := 1; i < len(keyvals); i += 2 {
			if _, ok := keyvals[i].(level.Value); ok {
				// overwriting the key without copying keyvals
				// techically violates the log.Logger contract
				// but is safe in this context because none
				// of the loggers in this program retain a reference
				// to keyvals
				keyvals[i-1] = key
				break
			}
		}
		return logger.Log(keyvals...)
	})
}
```
